### PR TITLE
Ex 1 2 and 3 Topic 2

### PR DESCRIPTION
**Pull request by julianpoma**
I made Ex 1,2 and 3 of the Topic 2: OOP and Inheritance.

### CODE REVIEW

File: 02-OOP-Inheritance/Ex1-3/Ex-1to3.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    off (eventName, callback) {
+        if (this.events[eventName]) {
+            this.events[eventName] = undefined;
+        }
+    }
+}
+
+class Logger {
+    constructor () {}
```
Comment (maximilianotulian) [inline 02-OOP-Inheritance/Ex1-3/Ex-1to3.js]: mmm, interesting, do we need to keep the constructor in this case?

File: 02-OOP-Inheritance/Ex1-3/Ex-1to3.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    log(info) {}
```
Comment (maximilianotulian) [inline 02-OOP-Inheritance/Ex1-3/Ex-1to3.js]: Let's add the needed method to showcase the info in browser console

File: 02-OOP-Inheritance/Ex1-3/Ex-1to3.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    constructor(title, year, duration) {
+        super();
+        this.title = title;
+        this.year = year;
+        this.duration = duration;
+        this.castList = [];
+    }
+
+    play () {
+        super.emit('play');
```
Comment (maximilianotulian) [inline 02-OOP-Inheritance/Ex1-3/Ex-1to3.js]: `super.methodName()` is fine, but I would rather use it only when is needed. Probably you've learned after the conversation in the meeting, not need to change it.

File: 02-OOP-Inheritance/Ex1-3/Ex-1to3.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    resume () {
+        super.emit('resume');
+    }
+
+    pause () {
+        super.emit('pause');
+    }
+
+    addCast (actors) {
+        if(typeof actors === 'Object')
```
Comment (maximilianotulian) [inline 02-OOP-Inheritance/Ex1-3/Ex-1to3.js]: **Minor**: add a blank space before `(`. 

Probably I've told you about the **Minor**. Is like a small suggestion, in this particular case related to the code standard I currently following.

File: 02-OOP-Inheritance/Ex1-3/Ex-1to3.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+class Actor {
+    constructor(name, age) {
+        this.name = name;
+        this.age = age;
+    }
+}
+
+let terminator = new Movie('Terminator I', 1985, 60);
+
+terminator.on('play', () => { console.log('Event play started'); });
```
Comment (maximilianotulian) [inline 02-OOP-Inheritance/Ex1-3/Ex-1to3.js]: The line `70` produces the expected result forexercise three. The problem is how we're getting the results. The callback provided should be the method log of a logger instance. Please take an additional look on it.

File: 02-OOP-Inheritance/Ex1-3/Ex-1to3.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
+class EventEmitter {
+    constructor() {
+        this.events = new Object();
```
Comment (julianpoma) [inline 02-OOP-Inheritance/Ex1-3/Ex-1to3.js]: Fixed!

File: 02-OOP-Inheritance/Ex1-3/Ex-1to3.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+        }
+    }
+
+    emit (eventName) {
+        this.events[eventName]();
+    }
+
+    off (eventName, callback) {
+        if (this.events[eventName]) {
+            this.events[eventName] = undefined;
```
Comment (julianpoma) [inline 02-OOP-Inheritance/Ex1-3/Ex-1to3.js]: Not the pair, but the eventname wont have any function attached to it

File: 02-OOP-Inheritance/Ex1-3/Ex-1to3.js (hunk 1 above)
Comment (julianpoma) [inline 02-OOP-Inheritance/Ex1-3/Ex-1to3.js]: No, its not needed

File: 02-OOP-Inheritance/Ex1-3/Ex-1to3.js (hunk 4 above)
Comment (julianpoma) [inline 02-OOP-Inheritance/Ex1-3/Ex-1to3.js]: The space is there in my code !

File: 02-OOP-Inheritance/Ex1-3/Ex-1to3.js (hunk 5 above)
Comment (julianpoma) [inline 02-OOP-Inheritance/Ex1-3/Ex-1to3.js]: Fixed :)

**Comment by julianpoma** [comment]
This wont be any longer needed. I put all the ex on the other branch and togheter

**Comment by maximilianotulian** [comment]
Fine, go a head and decline this pull request

**Pull request closed without merging**